### PR TITLE
fix(web-apps): Use Angular's i18n proposed implementation

### PR DESCRIPTION
**Pull request by kimwnasptd**
First part for https://github.com/kubeflow/kubeflow/issues/6042
closes https://github.com/kubeflow/kubeflow/issues/6041

This PR:
* Uses the `i18n` attribute and `$localize` function to mark text for translation in the frontend
* Creates distinct folders with OWNERS for the French language
* Removes the `ngx-translate` package
* Reverts the strings in the backend to be parsed by `ngx-translate`
* Updates READMEs and `package.json`/`angular.json` files with instructions on how to run and serve the French flavors locally 

@wg102 @Jose-Matsuda I added you as the owners for the French translations for each of the web app/s `i18n/fr/OWNERS` files, following your work in https://github.com/kubeflow/kubeflow/pull/5880. The translation files are not up to date though, so we will need to manually update them once more for this new implementation.

/cc @tasos-ale @StefanoFioravanzo 
/assign @kimwnasptd

### CODE REVIEW

**Comment by kimwnasptd** [comment]
@wg102 @Jose-Matsuda is seems that you are not members of the Kubeflow org, and thus I can't add you to owners files.

Can you make a small PR like this one to add yourselves to the members list https://github.com/kubeflow/internal-acls/pull/472? I can be your sponsor :slightly_smiling_face:

File: components/crud-web-apps/jupyter/README.md
Comment (DavidSpek) [inline components/crud-web-apps/jupyter/README.md:91]: Do I understand correctly that the web apps will not dynamically change their language based on the browser language that is set after merging this PR?

/cc @wg102

Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/README.md:91]: Correct. This will be the next step to discuss in the effort of using Angular's i18n approach [that's why I didn't close #6042 with this PR], which will require backend/manifests work as well https://angular.io/guide/i18n#deploy-multiple-locales.

Comment (DavidSpek) [reply components/crud-web-apps/jupyter/README.md:91]: To avoid a regression the backend and manifest changes should be implemented before merging this PR.

Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/README.md:91]: > To avoid a regression

This PR exists to fix regressions and bugs introduced from the initial PR that were not even brought up in the review, like 
1. The way the apps report errors https://github.com/kubeflow/kubeflow/issues/6041
2. Making the backend messages being readable only if they are parsed from the frontend
3. Locking other web apps [Katib, Models] to use `ngx-translate` in order to use the latest common library. Issues with `ngx-translate` are tracked here https://github.com/kubeflow/kubeflow/issues/6042

> the backend and manifest changes should be implemented before merging this PR.

This is bad practice especially for a huge work like internationalization that touches many parts of the app. This effort should happen in iterations by documenting each step along the way.

Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/README.md:91]: I get your point though about the regression, since after the initial PR the apps had some internationalization support provided, and thank you for raising it. 

But I want to unblock upgrading the other web apps to use the latest common code, which right now means that they should follow the `ngx-translate` way. That's why I want to move forward with this first step, to untie the two efforts and be able to track them in parallel.

Comment (DavidSpek) [reply components/crud-web-apps/jupyter/README.md:91]: The deployment and manifest changes needed to not cause a regression in these web apps will also be needed for the other web apps, and can be implemented completely independently from any upgrading or internationalization effort.

In fact, it is quite surprising a proper web server was never used to begin with, since it’s stated everywhere that you should use a web server like NGINX and not the use gunicorn to server the static files. 

Also, The other web apps haven’t been translated yet. As such, there is no reason they would need `ngx-translate`. The translation for those web apps can just as easily be done after the upgrade has been done, so this doesn’t block the upgrade in any way.

Comment (DavidSpek) [reply components/crud-web-apps/jupyter/README.md:91]: I’d also like to point out that your statement `that were not even brought up in the review`, and other similar statements you’ve made, come across as blaming the reviewers of that PR for the bugs or regressions that slipped through the review process. Statements like this are appropriate within an open source project, not very welcoming for contributors and likely breach the community guidelines / code of conduct.

I remember a large PR being merged despite my review comments indicating a lot of issues in the code, but I’ll create a dedicated issue and PR to track the effort to fix this.

Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/README.md:91]: > The other web apps haven’t been translated yet. As such, there is no reason they would need ngx-translate. The translation for those web apps can just as easily be done after the upgrade has been done, so this doesn’t block the upgrade in any way.

The common frontend code is using the `TranslateModule` from the `ngx-translate` module. I.e. main class for the [`BackendService`](https://github.com/kubeflow/kubeflow/blob/bf406cf03476a8238123b89957a8927ef0f53e43/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/services/backend/backend.service.ts#L28) expects a `TranslateService` to be injected in order to work. 

This means that, in order to update the other apps to use the latest common code, they will need to install and use `ngx-translate`, which is something we want to avoid https://github.com/kubeflow/kubeflow/issues/6042.

Comment (DavidSpek) [reply components/crud-web-apps/jupyter/README.md:91]: Still, the solution needed to avoid a regression can be implemented regardless of the use of `ngx-translate`.

File: components/crud-web-apps/common/backend/kubeflow/kubeflow/crud_backend/status.py
Comment (tasos-ale) [inline components/crud-web-apps/common/backend/kubeflow/kubeflow/crud_backend/status.py:17]: @kimwnasptd You can remove the `key` parameter since it is not used.

File: components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/form/name-namespace-inputs/name-input/name-input.component.html
Comment (tasos-ale) [inline components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/form/name-namespace-inputs/name-input/name-input.component.html:3]: You forgot to mark the placeholder for translation here.

File: components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-image/form-image.component.html
Comment (tasos-ale) [inline components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-image/form-image.component.html:131]: This placeholder also needs to be marked translation.

File: components/crud-web-apps/volumes/frontend/src/app/pages/form/form-default/form-default.component.html
Comment (tasos-ale) [inline components/crud-web-apps/volumes/frontend/src/app/pages/form/form-default/form-default.component.html:37]: This label needs to be marked for translation.

**Review by tasos-ale — CHANGES_REQUESTED** [review]

File: components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/form/name-namespace-inputs/name-input/name-input.component.html
Comment (kimwnasptd) [reply components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/form/name-namespace-inputs/name-input/name-input.component.html:3]: Pushed a commit

File: components/crud-web-apps/volumes/frontend/src/app/pages/form/form-default/form-default.component.html
Comment (kimwnasptd) [reply components/crud-web-apps/volumes/frontend/src/app/pages/form/form-default/form-default.component.html:37]: Pushed a commit

File: components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-image/form-image.component.html
Comment (kimwnasptd) [reply components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-image/form-image.component.html:131]: pushed a commit

**Comment by StefanoFioravanzo** [comment]
Thanks for the work @kimwnasptd! This should unblock the effort to update the other web apps as well to Angular 12.

/hold cancel
/lgtm
/approve

**Comment by javyxu** [comment]
hello, How can I switch languages using Angular's i18N in a similar way, example: http://127.0.0.1/?lang=en or http://127.0.0.1/?lang=fr?

**Pull request merged**
